### PR TITLE
Writeup SSO pages

### PR DESCRIPTION
**Pull request by hola-soy-milk**

### CODE REVIEW

File: content/4.auth/7.sso/1.index.md [hunk 1]
```diff
@@ -1,7 +1,95 @@
 ---
-description:
+description: Single Sign-On is a mechanism which allows to use external providers to login into systems.
 ---
 
 # Single Sign-On
 
+Single Sign-On (SSO) is a mechanism which allows to use external providers to login into systems. For example, you can use
+your Google or Facebook account to authenticate into systems without the need to create a new registration on those
+systems.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
Single Sign-On (SSO) is a mechanism which allows to use external providers to login into systems. For example, you can use
your Google or Facebook account to authenticate without the need to create a new registration.
```

Felt a bit wordy

File: content/4.auth/7.sso/1.index.md [hunk 2]
```diff
@@ -1,7 +1,95 @@
...
+
+In [Directus version 10.10.0](/releases/breaking-changes) the `cookie` mode has
+been replaced by the new `session` mode. The API still supports `cookie` mode logins for compatibility, however the Data
+Studio no longer supports `cookie` mode for logging in.
+
+::
+
+## Supported SSO mechanisms
+
+Directus supports four standard types of SSO mechanisms:
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
Directus supports four SSO mechanisms:
```

File: content/4.auth/7.sso/1.index.md [hunk 3]
```diff
@@ -1,7 +1,95 @@
...
+
+- [OpenID](https://openid.net/specs/openid-connect-core-1_0.html)
+- [OAuth 2.0](https://www.ietf.org/rfc/rfc6750.txt)
+- [LDAP](https://datatracker.ietf.org/doc/html/rfc4511)
+- [SAML](https://datatracker.ietf.org/doc/html/rfc7522)
+
+Here are the configurations allowed for each one: [SSO configuration](/configuration/auth-sso)
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]: Make this a callout type link

File: content/4.auth/7.sso/1.index.md [hunk 4]
```diff
@@ -1,7 +1,95 @@
...
+
+In order to use these mechanisms you need to create an application/configuration on your preferred external provider, set the environment variables to configure the external provider and optionally, set the environment variables to configure cookies.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
In order to use these mechanisms you need to create an application/configuration on your preferred external provider, set the environment variables to configure the external provider and, optionally, set the environment variables to configure cookies.
```

File: content/4.auth/7.sso/1.index.md [hunk 5]
```diff
@@ -1,7 +1,95 @@
...
+
+## SSO with Directus behind Proxy
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
## SSO with Directus Behind a Proxy
```

File: content/4.auth/7.sso/1.index.md [hunk 6]
```diff
@@ -1,7 +1,95 @@
...
+
+If Directus is running behind an HTTP(S) proxy, the instance might not be able to reach the configured SSO provider. In
+such a case, you may want to use the [`global-agent`](https://www.npmjs.com/package/global-agent) package, allowing you
+to configure the corresponding proxy settings.
+
+In this quick guide, we'll show how to set up the `global-agent` package by extending the Directus Docker Image.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
In this guide, you'll learn how to set up the `global-agent` package by extending the Directus Docker Image.
```

you/yours, not we/ours for docs

File: content/4.auth/7.sso/1.index.md [hunk 7]
```diff
@@ -1,7 +1,95 @@
...
+
+::callout{type="warning" title="Security Concerns"}
+
+Due to the fact that the `global-agent` package needs to intercept all external requests, it can be regarded as a
+potential attack surface. Especially in critical environments, it's therefore recommended to thoroughly evaluate the
+impact of such a setup beforehand.
+
+::
+
+First, create a patch file to adjust the `pm2` configuration file so that `global-agent` is registered at startup:
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
Create a patch file to adjust the `pm2` configuration file so that `global-agent` is registered at startup:
```

File: content/4.auth/7.sso/1.index.md [hunk 8]
```diff
@@ -1,7 +1,95 @@
...
+ 		script: 'cli.js',
+ 		args: ['start'],
++		node_args: ['-r', 'global-agent/bootstrap'],
+
+ 		// General
+ 		instances: process.env.PM2_INSTANCES ?? 1,
+```
+
+Afterwards, in the same directory, create a `Dockerfile`. In there, we extend from the Directus Image, install the
+`global-agent` package and apply the previously created patch file:
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
In the same directory, create a `Dockerfile`. Extend the Directus Image, install the
`global-agent` package, and apply the previously created patch file:
```

Remove these filler words at the start, like 'first', 'next', etc. Always remain concise by just giving instruction instead of prefacing each sentence. Also we love an Oxford comma.

File: content/4.auth/7.sso/1.index.md [hunk 9]
```diff
@@ -1,7 +1,95 @@
...
+
+```Dockerfile-vue [Dockerfile]
+FROM directus/directus:{{ packages.directus.version.major }}.x.y
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]: This dynamic variable was in the old docs but I don't think it's available (yet) in the new docs. Hardcode the value and then make a note after the snippet.

File: content/4.auth/7.sso/1.index.md [hunk 10]
```diff
@@ -1,7 +1,95 @@
...
+USER node
+```
+
+A new Docker Image can now be built from this customized `Dockerfile`, and can then be used with the following
+environment variables to control the proxy configuration:
+
+- `GLOBAL_AGENT_HTTP_PROXY`
+- `GLOBAL_AGENT_HTTPS_PROXY`
+
 <ListChildren />
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]: Replace this with an explicit link please

File: content/4.auth/7.sso/2.seamless.md [hunk 11]
```diff
@@ -3,3 +3,122 @@ description:
 ---
 
 # Seamless SSO
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md:5]: Description please

File: content/4.auth/7.sso/2.seamless.md [hunk 12]
```diff
@@ -3,3 +3,122 @@ description:
...
+
+While sometimes you want your users to directly have access to the Directus project, in other cases you may need to
+fetch private data from Directus in your client using external providers. For such cases, a special configuration is
+required to work across domains.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
```suggestion
While sometimes you want your users to directly have access to the Directus project, in other cases you may need to
fetch private data from Directus in your application using external providers. 
```

> For such cases, a special configuration is required to work across domains.

This is implied.

File: content/4.auth/7.sso/2.seamless.md [hunk 13]
```diff
@@ -3,3 +3,122 @@ description:
...
+
+## Implementing Seamless SSO
+
+Setup an external provider. You'll find some examples further down below. Allow the cookie to be accessible across domains. There are two authentication mechanisms for this.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
```suggestion
Set up an external provider. You'll find some examples further down below. Allow the cookie to be accessible across domains. There are two authentication mechanisms for this.
```

I'm not too sure about this opener. I don't think it's clear.

File: content/4.auth/7.sso/2.seamless.md [hunk 14]
```diff
@@ -3,3 +3,122 @@ description:
...
+**Authentication Mode: session**
+
+```sh
+AUTH_<PROVIDER>_MODE="session"
+SESSION_COOKIE_DOMAIN="XXXX"
+SESSION_COOKIE_SECURE="true"
+SESSION_COOKIE_SAME_SITE="None"
+```
+
+**Authentication Mode: cookie (legacy)**
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]: If it's legacy, I'd vote we remove it?

File: content/4.auth/7.sso/2.seamless.md [hunk 15]
```diff
@@ -3,3 +3,122 @@ description:
...
+
+```html
+<a
+	href="https://directus.myserver.com/auth/login/google?redirect=https://client.myserver.com/login"
+	>Login</a
+>
+```
+
+In the above, `https://directus.myserver.com` should be the address of your Directus instance, while `https://client.myserver.com/login` should be the address of your client application. The `/login` path is not necessary, but helps to separate concerns.
+
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
```suggestion
Replace `https://directus.myserver.com` with the address of your Directus instance, while `https://client.myserver.com/login` should be the address of your client application. The `/login` path is not necessary, but helps to separate concerns.

```

Another example of sentence redundant openers

File: content/4.auth/7.sso/2.seamless.md [hunk 16]
```diff
@@ -3,3 +3,122 @@ description:
...
+```js
+import { createDirectus, authentication } from "@directus/sdk";
+
+const client = createDirectus("https://directus.myserver.com").with(
+	authentication("cookie", { credentials: "include" })
+);
+
+await client.refresh();
+```
+
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]: Whichever snippets are kept - whether we remove cookie or not - put this in a code tab. Look at the authentication partial for an example.

File: content/4.auth/7.sso/2.seamless.md [hunk 17]
```diff
@@ -3,3 +3,122 @@ description:
...
+**Authentication Mode: cookie**
+
+```sh
+REFRESH_TOKEN_COOKIE_SECURE="false"
+REFRESH_TOKEN_COOKIE_SAME_SITE="lax"
+```
+
+Note that no `REFRESH_TOKEN_COOKIE_DOMAIN` or `SESSION_COOKIE_DOMAIN` value is set.
+
+::callout{type="warning" title="Disabling secured cookies"}
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
```suggestion
::callout{type="warning" title="Disabling Secured Cookies"}
```

**Review by phazonoverload — CHANGES_REQUESTED** [review]
Good start. Very similar feedback throughout - pronouns are weird, lots of redundant sentence starters, and a desire to use tabs for equivalent examples with the API/SDK (although elsewhere in the docs we just show the REST examples - look at the 2FA page - maybe we do the same here?).

File: content/4.auth/7.sso/2.seamless.md (hunk 14 above)
Comment (hola-soy-milk) [reply content/4.auth/7.sso/2.seamless.md]: I'm so glad you agree, gonna joyfully remove

File: content/4.auth/7.sso/1.index.md (hunk 10 above)
Comment (hola-soy-milk) [reply content/4.auth/7.sso/1.index.md]: This being the `ListChildren` component?

Comment (hola-soy-milk) [reply content/4.auth/7.sso/1.index.md]: Because if so, we have the "Next page" links below that would do the same

**Comment by hola-soy-milk** [comment]
Should be sorted, thanks for the review! Could you please take another look?

File: content/4.auth/7.sso/1.index.md [hunk 18]
```diff
@@ -1,7 +1,97 @@
...
+Directus supports four SSO mechanisms:
+
+- [OpenID](https://openid.net/specs/openid-connect-core-1_0.html)
+- [OAuth 2.0](https://www.ietf.org/rfc/rfc6750.txt)
+- [LDAP](https://datatracker.ietf.org/doc/html/rfc4511)
+- [SAML](https://datatracker.ietf.org/doc/html/rfc7522)
+
+::callout{type="info" url="/configuration/auth-sso"}
+
+Find out what the configurations for each SSO configuration are.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
Check out specific the SSO configuration variables for different mechanisms.
```

File: content/4.auth/7.sso/1.index.md [hunk 19]
```diff
@@ -1,7 +1,97 @@
...
+Directus supports four SSO mechanisms:
+
+- [OpenID](https://openid.net/specs/openid-connect-core-1_0.html)
+- [OAuth 2.0](https://www.ietf.org/rfc/rfc6750.txt)
+- [LDAP](https://datatracker.ietf.org/doc/html/rfc4511)
+- [SAML](https://datatracker.ietf.org/doc/html/rfc7522)
+
+::callout{type="info" url="/configuration/auth-sso"}
+
+Check out specific the SSO configuration variables for different mechanisms.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
Check out specific SSO configuration variables for different mechanisms.
```

File: content/4.auth/7.sso/1.index.md [hunk 20]
```diff
@@ -1,7 +1,97 @@
...
+
+::callout{type="info" title="Session Based Authentication"}
+
+In [Directus version 10.10.0](/releases/breaking-changes) the `cookie` mode has
+been replaced by the new `session` mode. The API still supports `cookie` mode logins for compatibility, however the Data
+Studio no longer supports `cookie` mode for logging in.
+
+::
+
+## Supported SSO mechanisms
```
Comment (phazonoverload) [inline content/4.auth/7.sso/1.index.md]:
```suggestion
## Supported SSO Mechanisms
```

File: content/4.auth/7.sso/2.seamless.md [hunk 21]
```diff
@@ -1,5 +1,106 @@
...
+::
+
+::callout{type="info" title="Redirect Allow List"}
+
+To allow Directus to redirect to external domains like `https://client.myserver.com/` used above, you'll need to include
+it in the `AUTH_<PROVIDER>_REDIRECT_ALLOW_LIST` security setting.
+
+::
+
+## Testing Seamless SSO locally
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
```suggestion
## Testing Seamless SSO Locally
```

**Review by phazonoverload — APPROVED** [review]
LGTM

File: content/4.auth/7.sso/2.seamless.md [hunk 22]
```diff
@@ -1,5 +1,106 @@
...
+```
+
+**Authentication Mode: cookie**
+
+```sh
+REFRESH_TOKEN_COOKIE_SECURE="false"
+REFRESH_TOKEN_COOKIE_SAME_SITE="lax"
+```
+
+Note that no `REFRESH_TOKEN_COOKIE_DOMAIN` or `SESSION_COOKIE_DOMAIN` value is set.
```
Comment (phazonoverload) [inline content/4.auth/7.sso/2.seamless.md]:
````suggestion
Locally, Directus won't be served under
a valid SSL certificate, which is a requirement for "Secure" cookies. Instead, for local testing purposes (and local
testing purposes only), the following configuration can be used when using session authentication:

```sh
SESSION_COOKIE_SECURE="false"
SESSION_COOKIE_SAME_SITE="lax"

```
Note that no  `SESSION_COOKIE_DOMAIN` value is set.
````
